src/Symfony/Component/RateLimiter/RateLimiterFactory.php

```diff
@@ -69,7 +69,11 @@ protected static function configureOptions(OptionsResolver $options): void
     {
         $intervalNormalizer = static function (Options $options, string $interval): \DateInterval {
             try {
-                return (new \DateTimeImmutable())->diff(new \DateTimeImmutable('+'.$interval));
+                // Force UTC timezone, because we don't want to deal with quirks happening when modifying dates in case
+                // there is a default timezone with DST. Read more here https://github.com/symfony/symfony/pull/58757
+                $now = new \DateTimeImmutable('now', new \DateTimeZone('UTC'));
```

Comment:
doing it this way will allow using `group time-sensitive` to mock the clock:
```suggestion
                $now = \DateTimeImmutable::createFromFormat('U', time(), new \DateTimeZone('UTC'));
```